### PR TITLE
Ato 578/create spot response lambda

### PR DESCRIPTION
**Pull request by CarlyG55**
## What

Create spot response lambda, also add flag for back channel logout queue
## Checklist

- [ ] Impact on orch and auth mutual dependencies has been checked.

- [ ] A UCD review has been performed.

### CODE REVIEW

**Comment by CarlyG55** [comment]
This will need to be tested in staging

File: ci/terraform/oidc/backchannel-logout-request.tf [hunk 1]
```diff
@@ -71,6 +71,7 @@ resource "aws_lambda_alias" "backchannel_logout_request_lambda_active" {
 }
 
 resource "aws_lambda_event_source_mapping" "backchannel_logout_lambda_sqs_mapping" {
+  count            = var.orch_backchannel_logout_enabled ? 0 : 1
```
Comment (hannah-leong) [inline ci/terraform/oidc/backchannel-logout-request.tf:74]: Should this be the other way around?
```suggestion
  count            = var.orch_backchannel_logout_enabled ? 1 : 0
```

File: ci/terraform/oidc/spot-response.tf [hunk 2]
```diff
@@ -61,7 +61,7 @@ resource "aws_iam_policy" "spot_response_sqs_read_policy" {
 }
 
 resource "aws_lambda_event_source_mapping" "spot_response_lambda_sqs_mapping" {
-  count            = var.spot_enabled ? 1 : 0
+  count            = var.spot_enabled && !var.orch_spot_response_enabled ? 1 : 0
   event_source_arn = aws_ssm_parameter.spot_response_queue_arn.value
   function_name    = aws_lambda_function.spot_response_lambda.arn
   batch_size       = 1
```
Comment (hannah-leong) [inline ci/terraform/oidc/spot-response.tf:67]: Why is this `!var.orch_spot_response_enabled`? Is the flag for if we are mocking it or not?

File: ci/terraform/oidc/variables.tf [hunk 3]
```diff
@@ -535,6 +535,18 @@ variable "orch_authorisation_enabled" {
...
+  type        = bool
+  default     = false
+}
+
+variable "orch_spot_response_enabled" {
+  description = "Flag to disable routing spot response traffic to the authentication account"
+  type        = bool
+  default     = false
+}
+
```
Comment (hannah-leong) [inline ci/terraform/oidc/variables.tf:549]: I find having them be `flag_enabled` with a description that says "disable" a bit confusing - maybe switch it round? Think it is more consistent with the other flags too
```suggestion
variable "orch_backchannel_logout_enabled" {
  description = "Flag to enable routing back channel logout traffic to the authentication account"
  type        = bool
  default     = false
}

variable "orch_spot_response_enabled" {
  description = "Flag to enable routing spot response traffic to the authentication account"
  type        = bool
  default     = false
}

```

Comment (hannah-leong) [reply ci/terraform/oidc/variables.tf:549]: Unless these are disable flags in which case maybe change the names to `flag_disabled`

**Review by hannah-leong — DISMISSED** [review]

**Comment by hannah-leong** [comment]
Let me know when conflicts are resolved/ready for re-approval :)

**Pull request closed without merging**